### DISCUSSION TITLE
bind_enabled_from doesn't seem to work for me

### DISCUSSION THREAD

**Discussion by skullcrash** [zauberzeug/nicegui#2621] [Q&A] [answered]
### Question

If I execute this code : 
```py
test = ui.input(label='Test',
        validation={
        'char : 2 !': lambda value: len(value) == 2,
        'char : 2 !': lambda value: len(value) > 1}) 
c = ui.button('Continue',on_click=lambda: (ui.open('/contact'))
c.bind_enabled_from(test, 'error')
```
The button is disabled at start.
After typing one char, it is enabled.
After typing 2 char, it is disabled again
After typing 3 char and more, it stays disabled.
Is it not supposed to behave the opposite way ?

**Comment by falkoschindler** [comment, marked as answer]
Hi @skullcrash,

Your dictionary has duplicate entries.
And the lambda expression should return `True` for the "valid" case.
And the binding should convert the `error` property in a boolean, e.g. return `False` when the error is not `None`.

Here is an example:
```py
test = ui.input(label='Test', validation={
    'Should not be two characters long!': lambda value: len(value) != 2,
    'Should be longer than one character!': lambda value: len(value) > 1,
})
c = ui.button('Continue', on_click=lambda: ui.open('/contact'))
c.bind_enabled_from(test, 'error', lambda error: error is None)
```

**Reply by skullcrash** [reply]
Thank you @falkoschindler  ! It is more clear to me now. I keep searching though how to show the button 'disabled', at the beginning, before entering any value in the "test" input field. At the beginning there is no error yet. 
The following code doesn't seem to work in the context of the code above : `c.disable()`

**Reply by skullcrash** [reply]
I think I found the solution. This seems to work the way I want :
```py
c = ui.button('Continue', on_click=lambda: ui.open('/contact')).props("disable")
c.bind_enabled_from(test, 'error', lambda error: error is None)
```
